src/backend/tcop/zenith_wal_redo.c

```diff
@@ -98,6 +98,11 @@ static ssize_t buffered_read(void *buf, size_t count);
 
 static BufferTag target_redo_tag;
 
+/*
+ * Buffer with target WAL redo page.
+ * We need to pin this buffer i memory but we c an not just call ReadBuffer because
+ * in some contexts it is assumed that buffer is not locked.
```

Comment:
```suggestion
 * Buffer with target WAL redo page.
 * We must not evict this page from the buffer pool, but we cannot just keep it pinned because
 * some WAL redo functions expect the page to not be pinned. So we have a special check in
 * localbuf.c to prevent this buffer from being evicted.
```